### PR TITLE
GraphicBuilder.addCurvePrimitive.

### PR DESCRIPTION
**Pull request by pmconne**

### CODE REVIEW

File: common/api/geometry-core.api.md [hunk 1]
```diff
@@ -183,6 +183,9 @@ export type AnnounceNumberNumberCurvePrimitive = (a0: number, a1: number, cp: Cu
 // @public
 export type AnyCurve = CurvePrimitive | CurveCollection;
 
+// @public
+export type AnyCurvePrimitive = Arc3d | LineSegment3d | LineString3d | BSplineCurve3d | BezierCurve3d | DirectSpiral3d | IntegratedSpiral3d | CurveChainWithDistanceIndex;
```
Comment (bbastings) [inline common/api/geometry-core.api.md:187]: CurveChainWithDistanceIndex is a CurveChain already, not sure what tryAddPath does...I probably would leave this out?

**Review by markschlosseratbentley — APPROVED** [review]

**Review by bbastings — APPROVED** [review]

File: common/api/geometry-core.api.md (hunk 1 above)
Comment (pmconne) [reply common/api/geometry-core.api.md:187]: No, it is a subclass of CurvePrimitive that has a CurveChain as a member.
Path.tryAddChild will append any it.

Comment (bbastings) [reply common/api/geometry-core.api.md:187]: Ok, wonder if anyone will ever use this. :)

**Comment by pmconne** [comment]
Waiting for #1733 which adds a new CurvePrimitiveType.

**Comment by bbastings** [comment]
Is there a CurvePrimitive type that wouldn't work with tryAddPath? i.e. Why not have the arg be CurvePrimitive instead of AnyCurvePrimitive? InterpolationCurve will be handled by the tryAddPath case...

**Comment by pmconne** [comment]
@bbastings The switch statement would not work with `CurveVector`. `AnyCurveVector` is to `CurveVector` as `AnyGeometryQuery` is to `GeometryQuery`.

**Comment by bbastings** [comment]
@pmconne Not as written no, but it could do this: switch (curve.curvePrimitiveType) { case CurvePrimitiveType.Arc: (curve as Arc3d).doArc3dStuff();

**Pull request merged**